File: javascript/series/series.js
class Series {

    constructor(digits) {
        this.digits = [...digits].map(Number);
    }

    slices(size) {
        if (size > this.digits.length) {
            throw new Error('Slice size is too big.');
        }
        return [...Array(this.digits.length - size + 1).keys()]
            .map(idx => this.digits.slice(idx, idx + size));
    }

}

module.exports = Series;

File: fsharp/series/Series.fs
module Series

let slices (str: string) (length: int): string list option = 
    if str.Length < 1 || length < 1 || length > str.Length
    then None
    else
        [0 .. str.Length - length]
        |> List.mapi (fun n i -> str.[i .. i + length - 1])
        |> Some